### PR TITLE
Jamba: update integration tests

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

🟢 Fixes `generate`-related integration tests for jamba 🟢 

I've checked them against:
- my machine (compute capability 8); 
- our slow CI (compute capability 7) -- CI is still red, but due to a SDPA test.

⚠️ skips logits checks on older devices: there are big differences across different versions, possibly explained by custom cuda kernels. Given the relatively low usage of jamba, we don't have the bandwidth to dive.

_________________________

Detective work 🕵️ 
1. The tests use a [dummy model](https://huggingface.co/ai21labs/Jamba-tiny-random), `ai21labs/Jamba-tiny-random`, the generation text quality doesn't matter.
2. On my machine, checking out to the [commit](https://github.com/huggingface/transformers/pull/29943/commits/a3e8094af98690f149320a9cf3b7d9cc21dd50ea) that added the current tests, I already get a difference. My machine has the same major compute capability as an A100 (RTX4090).
3. We can see in the original PR that the tests were being updated as changes were made. There were a few commits after the latest version of the tests that modify the outputs, which would explain the failing tests since day 1

### CODE REVIEW

**Comment by gante** [comment]
cc @ydshieh

File: tests/models/jamba/test_modeling_jamba.py [hunk 1]
```diff
@@ -650,82 +650,112 @@ def test_new_cache_format(self, num_beams, do_sample):
 class JambaModelIntegrationTest(unittest.TestCase):
     model = None
     tokenizer = None
+    # This variable is used to determine which CUDA device are we using for our runners (A10 or T4)
+    # Depending on the hardware we get different logits / generations
+    cuda_compute_capability_major_version = None
```
Comment (gante) [inline tests/models/jamba/test_modeling_jamba.py:655]: This `cuda_compute_capability_major_version` pattern is copied from other models like e.g. `gemma`

**Comment by ydshieh** [comment]
(thank you for trigger the tests on the runner 🙏 )

File: tests/models/jamba/test_modeling_jamba.py [hunk 2]
```diff
@@ -650,82 +650,112 @@ def test_new_cache_format(self, num_beams, do_sample):
...
+                    0.1729, -0.4609, -0.0410,  0.2412,  0.1572, -0.1895,  0.2402, -0.0583
+                ]
+                , dtype=torch.float32)  # fmt: skip
+
+            EXPECTED_LOGITS_NO_GRAD_1 = torch.tensor(
+                [
+                -0.1318,  0.2354, -0.4160, -0.0325, -0.0461,  0.0342,  0.2578,  0.0874,
+                    0.1484,  0.2266, -0.1182, -0.1396, -0.1494, -0.1089, -0.0019, -0.2852,
+                    0.1973, -0.2676,  0.0586, -0.1992, -0.2520, -0.1147, -0.1973,  0.2129,
+                    0.0520,  0.1699,  0.1816,  0.1289,  0.1699, -0.1216, -0.2656, -0.2891,
```
Comment (amyeroberts) [inline tests/models/jamba/test_modeling_jamba.py:755]: ultra ultra nit 

```suggestion
                    0.0166, -0.2227,  0.0396, -0.1035,  0.0459,  0.2754, -0.1445,  0.1641,
                  -0.2910, -0.0273,  0.0227, -0.5547, -0.2139, -0.1396, -0.1582,  0.1289,
                    0.0713,  0.2256,  0.1699, -0.2295, -0.1182, -0.1167, -0.1387,  0.0261,
                    0.1270,  0.2285,  0.0403,  0.1108, -0.1318, -0.2334,  0.1455, -0.3945,
                    0.1729, -0.4609, -0.0410,  0.2412,  0.1572, -0.1895,  0.2402, -0.0583
                ]
                , dtype=torch.float32)  # fmt: skip

            EXPECTED_LOGITS_NO_GRAD_1 = torch.tensor(
                [
                   -0.1318,  0.2354, -0.4160, -0.0325, -0.0461,  0.0342,  0.2578,  0.0874,
                    0.1484,  0.2266, -0.1182, -0.1396, -0.1494, -0.1089, -0.0019, -0.2852,
                    0.1973, -0.2676,  0.0586, -0.1992, -0.2520, -0.1147, -0.1973,  0.2129,
                    0.0520,  0.1699,  0.1816,  0.1289,  0.1699, -0.1216, -0.2656, -0.2891,
```

File: tests/models/jamba/test_modeling_jamba.py [hunk 3]
```diff
@@ -650,82 +650,112 @@ def test_new_cache_format(self, num_beams, do_sample):
...
-                0.0703,  0.2236,  0.1729, -0.2285, -0.1152, -0.1177, -0.1367,  0.0289,
-                0.1245,  0.2363,  0.0442,  0.1094, -0.1348, -0.2295,  0.1494, -0.3945,
-                0.1777, -0.4570, -0.0408,  0.2412,  0.1562, -0.1943,  0.2373, -0.0593
-            ]
-            , dtype=torch.float32)  # fmt: skip
+            EXPECTED_LOGITS_NO_GRAD = torch.tensor(
+                [
+                    0.0134, -0.2197,  0.0396, -0.1011,  0.0459,  0.2793, -0.1465,  0.1660,
+                -0.2930, -0.0278,  0.0269, -0.5586, -0.2109, -0.1426, -0.1553,  0.1279,
+                    0.0713,  0.2246,  0.1660, -0.2314, -0.1187, -0.1162, -0.1377,  0.0292,
```
Comment (amyeroberts) [inline tests/models/jamba/test_modeling_jamba.py:696]: another tiny tiny nit 

```suggestion
                    0.0134, -0.2197,  0.0396, -0.1011,  0.0459,  0.2793, -0.1465,  0.1660,
                  -0.2930, -0.0278,  0.0269, -0.5586, -0.2109, -0.1426, -0.1553,  0.1279,
                    0.0713,  0.2246,  0.1660, -0.2314, -0.1187, -0.1162, -0.1377,  0.0292,
```

**Review by amyeroberts — APPROVED** [review]
Thanks for digging into this and fixing, and for writing a detailed PR description ❤️ 

Agreed it's not worth digging into given jamba usage, and as the generated texts appear similar despite the logic differences

File: tests/models/jamba/test_modeling_jamba.py [hunk 4]
```diff
@@ -650,82 +650,112 @@ def test_new_cache_format(self, num_beams, do_sample):
...
-                0.2002, -0.2676,  0.0598, -0.1982, -0.2539, -0.1133, -0.1973,  0.2148,
-                0.0559,  0.1670,  0.1846,  0.1270,  0.1680, -0.1250, -0.2656, -0.2871,
-                0.2344,  0.2637,  0.0510, -0.1855,  0.2158, -0.1289,  0.1758,  0.0074
-            ]
-            , dtype=torch.float32)  # fmt: skip
-
-        torch.testing.assert_close(logits[0, -1, :40].cpu(), EXPECTED_LOGITS_NO_GRAD_0, rtol=1e-3, atol=1e-3)
-        torch.testing.assert_close(logits[1, -1, :40].cpu(), EXPECTED_LOGITS_NO_GRAD_1, rtol=1e-3, atol=1e-3)
+        # TODO: there are significant differences in the logits across major cuda versions, which shouldn't exist
+        if self.cuda_compute_capability_major_version == 8:
```
Comment (ydshieh) [inline tests/models/jamba/test_modeling_jamba.py:736]: maybe better to use 

> self.skipTest(reason="Skipping for T4 runners because ...")

Comment (gante) [reply tests/models/jamba/test_modeling_jamba.py:736]: oops, merged before seeing this comment! 

You have a good point, in fact we should split the test in two to test (/skip) the logits separately
